### PR TITLE
resolver: Himmelblau needs old token for refresh

### PR DESCRIPTION
**Pull request by dmulder**
Himmelblau needs access to the old token during
a refresh otherwise the GECOS is lost (AAD
responds with everything we need except GECOS).

Fixes #

Checklist

- [x] This pr contains no AI generated code
- [ ] cargo fmt has been run
- [ ] cargo clippy has been run
- [ ] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

**Comment by dmulder** [comment]
I was unable to run make test because master fails to build for me:
```
error[E0599]: no associated item named `NO_CHECK_TIME` found for struct `X509VerifyFlags` in the current scope
    --> /home/dmulder/.cargo/registry/src/github.com-1ecc6299db9ec823/webauthn-rs-core-0.4.9/src/attestation.rs:1277:49
     |
1277 |             .set_flags(verify::X509VerifyFlags::NO_CHECK_TIME)
     |                                                 ^^^^^^^^^^^^^
     |                                                 |
     |                                                 associated item not found in `X509VerifyFlags`
     |                                                 help: there is an associated constant with a similar name: `USE_CHECK_TIME`
```

**Comment by yaleman** [comment]
They both should be built into a default rust install:

* [cargo clippy](https://doc.rust-lang.org/stable/clippy/usage.html) is the Rust linter.
* [cargo fmt](https://github.com/rust-lang/rustfmt) is the code formatter.

That OpenSSL error is typically related to not having the proper libraries installed - try `cargo clean` then a build - the CI passed so I'm going to merge this as-is.

File: unix_integration/src/resolver.rs [hunk 1]
```diff
@@ -393,7 +393,7 @@ where
         account_id: &Id,
         token: Option<UserToken>,
     ) -> Result<Option<UserToken>, ()> {
-        match self.client.unix_user_get(account_id).await {
+        match self.client.unix_user_get(account_id, token.clone()).await {
```
Comment (Firstyear) [inline unix_integration/src/resolver.rs:396]: Should be references to avoid the clone here. Can you do another pr to make `token: Option<&UserToken>`?
